### PR TITLE
fix(rs-drive-abci): Get rs-drive-abci to commit block 1

### PR DESCRIPTION
**Pull request by lklimek**
## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated relevant unit/integration/functional/e2e tests
- [ ] I have made corresponding changes to the documentation

**For repository code-owners and collaborators only**
- [ ] I have assigned this pull request to a milestone

### CODE REVIEW

**Review by QuantumExplorer — DISMISSED** [review]

File: packages/rs-drive-abci/src/abci/commit.rs [hunk 1]
```diff
@@ -26,6 +27,10 @@ impl Commit {
         quorum_type: QuorumType,
         chain_id: &str,
     ) -> Self {
+        // we need to "un-reverse" quorum hash, as it was reversed in [CleanedCommitInfo::try_from]
+        let mut quorum_hash = ci.quorum_hash.to_vec();
+        quorum_hash.reverse();
+
```
Comment (PastaPastaPasta) [inline packages/rs-drive-abci/src/abci/commit.rs:33]:
```suggestion
```

File: packages/rs-drive-abci/src/abci/commit.rs [hunk 2]
```diff
@@ -34,7 +39,7 @@ impl Commit {
                 block_id: Some(block_id.try_into().expect("cannot convert block id")),
                 height: height as i64,
                 round: ci.round as i32,
-                quorum_hash: ci.quorum_hash.to_vec(),
+                quorum_hash: quorum_hash,
```
Comment (PastaPastaPasta) [inline packages/rs-drive-abci/src/abci/commit.rs:42]:
```suggestion
                // we need to "un-reverse" quorum hash, as it was reversed in [CleanedCommitInfo::try_from]
                quorum_hash: ci.quorum_hash.iter().rev().cloned().collect(),
```

File: packages/rs-drive-abci/src/abci/handlers.rs [hunk 3]
```diff
@@ -261,10 +261,15 @@ where
...
+        if !block_state_info.matches_current_block(
+            height as u64,
+            round as u32,
+            block_hash.clone(),
+        )? {
             return Err(Error::from(AbciError::RequestForWrongBlockReceived(format!(
-                "received request for height: {} round: {}, expected height: {} round: {}",
-                height, round, block_state_info.height, block_state_info.round
+                "received request for height: {} round: {}, block: {};  expected height: {} round: {}, block: {}",
+                height, round,block_hash.to_hex(),
```
Comment (PastaPastaPasta) [inline packages/rs-drive-abci/src/abci/handlers.rs:271]:
```suggestion
                height, round, block_hash.to_hex(),
```

File: packages/rs-drive-abci/src/execution/engine.rs [hunk 4]
```diff
@@ -674,22 +673,24 @@ where
             round,
             block_header.core_chain_locked_height,
             block_header.proposer_pro_tx_hash,
-            hash,
+            hash.clone(),
```
Comment (PastaPastaPasta) [inline packages/rs-drive-abci/src/execution/engine.rs:676]: this is copyable
```suggestion
            hash,
```

**Review by PastaPastaPasta — CHANGES_REQUESTED** [review]
I have a few minor suggestions

File: packages/rs-drive-abci/src/abci/commit.rs [hunk 5]
```diff
@@ -94,16 +96,20 @@ impl Commit {
 
         //todo: maybe cache this to lower the chance of a hashing based attack (forcing the
         // same calculation each time)
+        let quorum_hash = &self.inner.quorum_hash[..]
+            .try_into()
+            .expect("invalid quorum hash length");
```
Comment (QuantumExplorer) [inline packages/rs-drive-abci/src/abci/commit.rs:101]: don't use expect. As a rule of thumb only use expect when there can be no doubt that something is the way it should be. No doubt meaning like the line above it. Also you can use it in tests.

File: packages/rs-drive-abci/src/abci/mimic.rs [hunk 6]
```diff
@@ -243,11 +243,12 @@ impl<'a, C: CoreRPCLike> AbciApplication<'a, C> {
 
         //if not in testing this will default to true
         if self.platform.config.testing_configs.block_signing {
+            let quorum_hash:[u8;32] = current_quorum.quorum_hash[..].try_into().expect("wrong quorum hash len");
```
Comment (QuantumExplorer) [inline packages/rs-drive-abci/src/abci/mimic.rs:246]: expect

**Review by QuantumExplorer — APPROVED** [review]

**Pull request merged**